src/validation/rules/LoneSchemaDefinition.js

```diff
@@ -35,25 +35,22 @@ export function LoneSchemaDefinition(
       oldSchema.getMutationType() ||
       oldSchema.getSubscriptionType());
 
-  const schemaNodes = [];
+  let schemaDefinitionsCount = 0;
   return {
     SchemaDefinition(node) {
       if (alreadyDefined) {
         context.reportError(
-          new GraphQLError(canNotDefineSchemaWithinExtension(), [node]),
+          new GraphQLError(canNotDefineSchemaWithinExtensionMessage(), node),
         );
         return;
       }
-      schemaNodes.push(node);
-    },
-    Document: {
-      leave() {
-        if (schemaNodes.length > 1) {
-          context.reportError(
-            new GraphQLError(schemaDefinitionNotAloneMessage(), schemaNodes),
-          );
-        }
-      },
+
+      if (schemaDefinitionsCount > 0) {
+        context.reportError(
+          new GraphQLError(schemaDefinitionNotAloneMessage(), node),
+        );
+      }
+      ++schemaDefinitionsCount;
```

Comment:
Now it generates separate errors on every duplicated schema definition.
It makes the order of errors to match the order of definitions in SDL.
Based on very similar code here:
https://github.com/graphql/graphql-js/blob/c847a45a0efc71f821d777da6121e32edde853e8/src/validation/rules/LoneAnonymousOperation.js#L36-L40